tests/unit/s2n_client_server_name_extension_test.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/extensions/s2n_client_server_name.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    const char *test_server_name = "github.com";
+
+    /* should_send */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        /* server_name not set -> don't send */
+        EXPECT_FALSE(s2n_client_server_name_extension.should_send(conn));
+
+        /* server_name empty -> don't send */
+        EXPECT_SUCCESS(s2n_set_server_name(conn, ""));
+        EXPECT_FALSE(s2n_client_server_name_extension.should_send(conn));
+
+        /* server_name set -> send! */
```

Comment:
what does `send!` mean here?

Reply:
It means I was excited that we would send :) I can remove it to avoid confusion